core/src/main/java/io/undertow/server/handlers/proxy/mod_cluster/ModClusterContainer.java

```diff
@@ -394,10 +395,22 @@ Context findNewNode(final VirtualHost.HostEntry entry) {
      *
      * @param domain   the load balancing domain, if known
      * @param jvmRoute the original jvmRoute
+     * @param entry    the resolved virtual host entry
      * @return the context, {@code null} if not found
-     * @oaram entry      the resolved virtual host entry
      */
-    Context findFailoverNode(final VirtualHost.HostEntry entry, final String domain, final String jvmRoute, final boolean forceStickySession) {
+    Context findFailoverNode(final VirtualHost.HostEntry entry, final String domain, final String session, final String jvmRoute, final boolean forceStickySession) {
+
+        // If configured, deterministically choose the failover target by calculating hash of the session ID modding by number of electable nodes
+        if (modCluster.isDeterministicFailover()) {
```

Comment:
Is this guaranteed to always get a live node? Or can it pick nodes in error state?